### PR TITLE
Update vendored RPC

### PR DESCRIPTION
**Pull request by akalin-keybase**
Use rpc.Server.Run() instead of RunAsync().

### CODE REVIEW

File: go/kex2/provisioner.go [hunk 1]
```diff
@@ -118,9 +118,10 @@ func (p *provisioner) pickFirstConnection() (err error) {
 		if err = srv.Register(prot); err != nil {
 			return err
 		}
-		if err = srv.Run(true); err != nil {
-			return err
-		}
+		go func() {
+			_ = srv.Run()
+			// TODO: do something with the error.
```
Comment (oconnor663) [inline go/kex2/provisioner.go]: Would it be easy to log it now, instead of doing nothing? Or maybe if the error is never supposed to happen, panic?

Comment (oconnor663) [inline go/kex2/provisioner.go]: Or is this the always-nil value from above? In that case, I'd say panic :)

**Comment by oconnor663** [comment]
It's hard for me to follow all the channels, but the vendoring looks good :-D

**Comment by oconnor663** [comment]
Let me know if you have time/interest to walk me through the other stuff.

**Comment by akalin-keybase** [comment]
Yeah why don't I talk you through it tomorrow? The more people familiar with this the better!

**Comment by oconnor663** [comment]
Thanks for the run-through. :ship:

**Pull request merged**